functions/src/verify.ts

```diff
@@ -0,0 +1,49 @@
+const tcn = require("tcn-node");
+
+function enumMemoTypeOf(memoType: number): string {
+    let res: string;
+    switch(memoType) { 
+        case 0: { 
+           res = 'CoEpiV1'; 
+           break; 
+        } 
+        case 1: { 
+           res = 'CovidWatchV1'; 
+           break; 
+        } 
+        default: { 
+           res = 'Reserved'; 
+           break; 
+        } 
+     };
+     return res;
+}
```

Comment:
Wouldn't it be easier to just do a Map lookup instead of making a switch statement?

Also I think unknown values should error. Setting them to Reserved (0xff) would mean we are changing the memo type value which means the signature would fail but that would be quite confusing. It's better to fail early.

```suggestion
const MEMO_TYPES = new Map([
  [0, "CoEpiV1"],
  [1, "CovidWatchV1"],
  [0xff, "Reserved"],
]);

function enumMemoTypeOf(memoType: number): string {
  const res = MEMO_TYPES.get(memoType);
  if (res === undefined) {
    throw new Error(
      `Unknown memo type: ${memoType}. Valid values: ${[...MEMO_TYPES.keys()]}`
    );
  }
  return res;
}
```

Reply:
The thing is all values from 1 to 255 are valid but currently unassigned. What do you think @madhavajay ? If we stick to TCN implementation we will strictly reject any values outside  0,1 and 255 and we can do this suggestion. Or should we do whichever for the current version and later update it ? 

Btw, regardless of which we still should expose the enum from tcn-node.

Reply:
Currently the Rust TCN implementation will not accept any values other than the ones in its Enum (0, 1, or 0xff) so we can't support other values anyway. But I will extend the Rust code to allow an "Other" option containing any `u8` value (0 - 255) to make it future proof.

Yes I agree we should expose the enum from tcn-node, that will be done in 0.4.0. I'm not sure if you want to wait for that release before finishing this PR though as it will also change the API.

Reply:
Yeah I had this exact issue in another project, thats the annoying thing about ints they dont really have good sentinels for invalid and the classic solution has been -1 but then thats a signed int so yeah... nightmare. I think all 255 enums should use the final position as BAD by default so 255 would be incorrect. Anyway, for now we can just block anything but 1 & 2 if the tcn lib doesnt allow it.